client/js/socket-events/more.js

```diff
@@ -16,6 +15,12 @@ socket.on("more", function(data) {
 	const scrollable = chan.closest(".chat");
 	const heightOld = chan.height();
 
+	// If there are no more messages to show, just hide the button and do nothing else
+	if (!data.messages.length) {
```

Comment:
Well, if we have 100 messages in a channel, and we receive a new message, the top messages gets removed from the client, doesn't it?